building/trusted_applications.rst

```diff
@@ -299,3 +305,66 @@ sequence:
             return TEE_ERROR_NOT_SUPPORTED;
         }
     }
+
+.. _TASign:
+
+Signing of TAs
+**************
+
+All :ref:`REE Filesystem Trusted Applications<ree_fs_ta>` need to be signed. The signature is verified by :ref:`optee_os` upon loading of the TA. Within the :ref:`optee_os` source is a directory ``keys``. The public part of ``keys/default_ta.pem`` will be compiled into the :ref:`optee_os` binary and the signature of each TA will be verified against this key upon loading.
+Currently ``keys/default_ta.pem`` must contain an RSA key.
+
+.. warning::
+
+   :ref:`optee_os` comes with a default **private** key in its source to facilitate easy development, testing, debugging and QA.
+   Never deploy an :ref:`optee_os` binary with this key in production. Instead replace this key as soon as possible with a public key and keep the private part of the key offline, preferably on an HSM.
+
+.. note::
+
+   Currently only a single key for signing TAs is supported by :ref:`optee_os`.
+
+TAs are signed using the ``sign.py`` script referenced from ``ta/mk/ta_dev_kit.mk`` in :ref:`optee_os`. Its default behavior is to sign a compiled TA binary and attach the signature to form a complete TA for deployment. For **offline** signing, a three-step process is required: In a first step a digest of the compiled binary has to be generated, in the second step this digest is signed offline using the private key and finally in the third step the binary and its signature are stitched together into the full TA.
```

Comment:
What about adding the three steps as a numbered list?
```rst
  1. A digest of the compiled binary has to be generated.
  2. The digest is signed offline using the private key.
  3. The binary and its signature are stitched together into the full TA.
```

Reply:
In the context of this paragraph the visual strength of a numbered list is too strong. pulling the readers eye from the initial part of the paragraph. I prefer the smooth running text here.

Reply:
Ok